requirements/typing.txt

```diff
@@ -1,5 +1,6 @@
 mypy==1.5.1
 torch==2.1.0
+colored
```

Comment:
@manascb1344 We can't add a dependency to Lightning like this. The rich package is optional for Lightning. Is there a different way to detect the color?


> The progress bar will now adjust its color depending on whether the terminal supports color and if it's a truecolor terminal.

Could you check whether this feature is already supported in rich itself? It feels like this is probably something that the package already handles, we just haven't configured it properly. I'm not familiar with this unfortunately.

Reply:
Also, isn't purple suitable both for light and dark?